new-frontend/src/app/guards/chosenCollaboration.guard.ts

```diff
@@ -2,16 +2,23 @@ import { inject } from '@angular/core';
 import { CanActivateFn, Router } from '@angular/router';
 import { routePaths } from '../routes';
 import { CHOSEN_COLLABORATION } from '../models/constants/sessionStorage';
+import { ResourceType, OperationType } from '../models/api/rule.model';
+import { PermissionService } from '../services/permission.service';
+import { CollaborationService } from '../services/collaboration.service';
 
-export function chosenCollaborationGuard(): CanActivateFn {
+export function chosenCollaborationGuard(resource?: ResourceType, operation?: OperationType): CanActivateFn {
   return async () => {
     const router: Router = inject(Router);
+    const permissionService: PermissionService = inject(PermissionService);
+    const collaborationService: CollaborationService = inject(CollaborationService);
 
     const chosenCollaboration = sessionStorage.getItem(CHOSEN_COLLABORATION);
     if (!chosenCollaboration) {
       router.navigate([routePaths.start]);
       return false;
     }
-    return true;
+
+    const collaboration = await collaborationService.getCollaboration(chosenCollaboration);
```

Comment:
Use the already loaded collaboration data in the ChosenCollaborationService. Loading data in a guard will cause this data to be loaded on every page view that uses this guard and slows the application unnecessarily. Especially if the needed data is already loaded in a service.